src/firefly/js/visualize/MouseReadoutCntlr.js

```diff
@@ -93,35 +131,33 @@ function reducer(state=initState(), action={}) {
     switch (action.type) {
         case CHANGE_LOCK_BY_CLICK:
             return {...state,lockByClick:action.payload.lockByClick};
-        case CHANGE_LOCK_UNLOCK_BY_CLICK:
-            return {...state,isLocked:action.payload.isLocked};
         case CHANGE_READOUT_PREFS:
-            const readoutPref = state.readoutPref;
-            const key = Object.keys(action.payload.readoutPref);
-            readoutPref[key]=action.payload.readoutPref[ key];
-            return {...state,readoutPref:{...state.readoutPref,...readoutPref}};
+            return {...state,readoutPref: {...state.readoutPref, ...action.payload.readoutPref}};
         default:
             return state;
     }
 }
 
+
 const initState= () =>
     ({
         lockByClick : false,
         isLocked: false,
         readoutPref :{
             imageMouseReadout1:'eqj2000hms',
             imageMouseReadout2: 'fitsIP',
```

Comment:
These need to be removed since you already defined them in following lines using the constants.

Reply:
oh yes... I missed that... thanks